content/docs/reconciliation.md

```diff
@@ -1,36 +1,38 @@
 ---
 id: reconciliation
-title: Reconciliation
+title: Reconciliação
 permalink: docs/reconciliation.html
 ---
 
-React provides a declarative API so that you don't have to worry about exactly what changes on every update. This makes writing applications a lot easier, but it might not be obvious how this is implemented within React. This article explains the choices we made in React's "diffing" algorithm so that component updates are predictable while being fast enough for high-performance apps.
+O React provê uma API declarativa, assim, você não precisa se preocupar em saber exatamente o que mudou aem cada atualização. Isso torna mais fácil a criação de aplicações, apesar de não ser obvia a forma como isso é implementado no React. Este artigo explica as escolhas que fizemos no algoritimo de "diffing" para que as atualizaçãos nos componentes sejam previsíveis e rapidas o suficiente para aplicações de alta performance.
 
-## Motivation {#motivation}
+## Motivação {#motivation}
 
-When you use React, at a single point in time you can think of the `render()` function as creating a tree of React elements. On the next state or props update, that `render()` function will return a different tree of React elements. React then needs to figure out how to efficiently update the UI to match the most recent tree.
+Quando utilizamos React, podemos imaginar a função `render()` como uma função responsável por gerar à árvore de Elementos React.
 
-There are some generic solutions to this algorithmic problem of generating the minimum number of operations to transform one tree into another. However, the [state of the art algorithms](http://grfia.dlsi.ua.es/ml/algorithms/references/editsurvey_bille.pdf) have a complexity in the order of O(n<sup>3</sup>) where n is the number of elements in the tree.
+Na proxima atualização de `state` ou `props`, a função `render()` retornará uma árvore de Elementos React diferente. Dessa forma, o React precisará descobrir como ele pode atualizar a UI para sincronizar com a arvore mais recente.
```

Comment:
also please keep this text in the same line as the above text to avoid breaking stuff if an update comes from the upstream repo